Add protections for the get_data utility

alurban opened the pull request:
This PR fixes a bug in the `get_data` utility by adding protections against nonexistent frame files when attempting to build a frame file cache, and falling back to `TimeSeries{Dict}.get` rather than `fetch` when that fails. Unit tests are also updated to reflect these changes.

Other changes include:
* Remove redundant `obs` keyword argument from `get_data`, and infer the observatory from the `frametype` (when applicable) using a regular expression
* Remove references to `obs` from a few scripts
* Simplify a couple of unit tests
* Update docstrings
* Misc. typo fix in `gwdetchar-conlog`

cc @duncanmmacleod

duncanmmacleod commented on gwdetchar/io/datafind.py:
```diff
@@ -178,23 +172,19 @@ def get_data(channel, start, end, obs=None, frametype=None, source=None,
         return series_class.read(
             source, channel, start=start, end=end, nproc=nproc,
             verbose=verbose, **kwargs)
     elif isinstance(channel, str):
```
it occurs to me that this should probably be

```python
        elif not isinstance(channel, (list, tuple))
```

to match the condition above that declares `series_class`

alurban replied:
thanks @duncanmmacleod, I've changed this in the latest commit. I also restored the logic that attempts to load a cache using `gwdatafind` because I remembered that it's _much_ simpler that way, but provided a lightweight wrapper around `gwdatafind.find_urls` to add some protection against non-existent frame files.

alurban commented:
@duncanmmacleod, I just saw your longer comment, sorry about that. Response below:

I prefer to have `gwdetchar.io.datafind.get_data` attempt to create its own file cache before falling back to `TimeSeries.get`, mainly because it then becomes possible to sieve the file cache against requested channels that actually exist in frame files. This makes `gwdetchar` tools extremely robust to changes in channel names, which is very common between observing runs.

For example, `gwdetchar-scattering` is written to analyze optical squeezer suspensions by default, but it will still run successfully (with default options) over O1 or O2 data when those squeezer optics didn't exist. Likewise for `gwdetchar-omega`, whose default channel list configuration changed several times just in the first few months of 2019.

duncanmmacleod commented on gwdetchar/io/datafind.py:
```diff
@@ -178,23 +179,22 @@ def get_data(channel, start, end, obs=None, frametype=None, source=None,
             verbose=verbose, **kwargs)
-    elif isinstance(channel, str):
-        return series_class.fetch(
+    except (HTTPError, TypeError):  # frame files not found
```
Can you explain when an `HTTPError` would be raised?

alurban replied:
@duncanmmacleod, the `HTTPError` is raised whenever `gwdatafind.find_urls` can't locate frame files (you get a `400 Bad Request`).

alurban replied:
I think that if you don't pass `ops` and the first letter of `frametype` is not a meaningful observatory code (e.g. for SenseMon frames), this will also lead to `HTTPError`. The `TypeError` case comes when you pass neither `frametype` nor `source`. In all these cases, the idea is that it would then fall back to `TimeSeries.get` with no protection against missing channels in frame files.

alurban replied:
@duncanmmacleod, I stand corrected, if you attempt to read SenseMon frames with `frametype='SenseMonitor_hoft_L1_M'` and `obs=None` then the result is an `IORegistryError: Format could not be identified.` Passing `frametype=None` falls back to `TimeSeries.get`, and passing both `frametype` and `obs` successfully uses `TimeSeries.read`.

alurban commented:
@duncanmmacleod, I've fixed this so that the `obs` kwarg is no longer needed, the observatory is now inferred from `frametype` (when it's passed) using a regular expression. Also, whenever neither `frametype` nor `source` are passed, the utility will fall back to `TimeSeries.get`.

I've also updated the docstring to [hopefully] make it clearer to the user which arguments do what, and when.
